boxsdk/client/client.py

```diff
@@ -1300,3 +1300,127 @@ def device_pinners(self, enterprise_id, direction=None, limit=None, marker=None,
             fields=fields,
             return_full_pages=False,
         )
+
+    def metadata_template(self, scope, template_key):
+        """
+        Initialize a :class:`MetadataTemplate` object with the given scope and template key.
+
+        :param scope:
+            The scope of the metadata template, e.g. 'enterprise' or 'global'
+        :type scope:
+            `unicode`
+        :param template_key:
+            The key of the metadata template
+        :type template_key:
+            `unicode`
+        :returns:
+            The metadata template object
+        :rtype:
+            :class:`MetadataTemplate`
+        """
+        return self.translator.get('metadata_template')(
+            session=self._session,
+            object_id='{0}/{1}'.format(scope, template_key),
```

Comment:
Thinking about this more, I feel even more strongly that we should make this change.

I think the current approach is going to cause us problems, now and in the future.

You can already see one problem forming in this PR, where there's a bunch of places in the code where we need to manually concat the scope and templateKey to form the object_id.

Another place we're going to run into problems is with POST and PUT operations where the body expects metadata template. If we naively do `template.object_id`, that might be incorrect if the API expects the actual id (I know of at least one API that was supposed to be doing this in the future). If we do `template.id`, that field might not actually be available, if it wasn't constructed with one.

For simplicity, we could make it so that if we're constructed with the id, we fetch the scope and templateKey, and also vice-versa, that way we always have all of those pieces of information available.